src/service/service.ts

```diff
@@ -83,18 +79,14 @@ class Service {
 }
 
 interface Tasks {
-  [task: string]: (inputs: TaskInputs, outputs: TaskOutputs) => void
+  [task: string]: (inputs: TaskInputs) => TaskOutputs
```

Comment:
Let's keep it simple and remove the `TaskOutputs` and just put
`object | Promise<object>`